src/windows/room/chat.rs

```diff
@@ -202,29 +203,42 @@ impl ChatState {
                         },
                     };
 
-                    if !force && filename.exists() {
-                        let msg = format!(
-                            "The file {} already exists; use :download! to overwrite it.",
-                            filename.display()
-                        );
-                        let err = UIError::Failure(msg);
+                    if filename.exists() {
+                        if !flags.intersects(DownloadFlags::FORCE | DownloadFlags::OPEN) {
+                            let msg = format!(
+                                "The file {} already exists; add ! to end of command to overwrite it.",
+                                filename.display()
+                            );
+                            let err = UIError::Failure(msg);
 
-                        return Err(err);
-                    }
+                            return Err(err);
+                        }
 
-                    let req = MediaRequest { source, format: MediaFormat::File };
+                        let req = MediaRequest { source, format: MediaFormat::File };
 
-                    let bytes =
-                        media.get_media_content(&req, true).await.map_err(IambError::from)?;
+                        let bytes =
+                            media.get_media_content(&req, true).await.map_err(IambError::from)?;
 
-                    fs::write(filename.as_path(), bytes.as_slice())?;
+                        fs::write(filename.as_path(), bytes.as_slice())?;
 
-                    msg.downloaded = true;
+                        msg.downloaded = true;
+                    }
 
-                    let info = InfoMessage::from(format!(
-                        "Attachment downloaded to {}",
-                        filename.display()
-                    ));
+                    let info = if flags.contains(DownloadFlags::OPEN) {
+                        // Contrary to the docs, open::that *is* blocking (on linux with xdg-open).
+                        // https://github.com/Byron/open-rs/issues/51#issuecomment-1408261825
+                        #[allow(deprecated)]
+                        open::that_in_background(&filename);
```

Comment:
I think you could use `open::that` here to avoid the deprecated function by doing:

```rust
let target = filename.to_os_string();
let _ = tokio::spawn_blocking(move || open::that_in_background(target));
```

tokio will then make a new blocking thread for the call, until the configured limit on the number of blocking threads is reached (512 by default).

Reply:
allright, clippy told me to explicitly `std::mem::drop` the `_` here. I am not sure, if you think it's not necessary then I'd remove it.